File: 2023-I/corte2/4-Arboles/ArbolesJava/Arbol.java
/*
 * Carlos A Delgado
 * 27 de Mayo de 2023
 * Implementación de un arbol
 */

public class Arbol{
  private int valor;
  private Arbol hizq;
  private Arbol hder;

  public Arbol(int valor, Arbol hizq, Arbol hder) {
    this.valor = valor;
    this.hizq = hizq;
    this.hder = hder;
  }

  public String preorden(){
    String recorrIzq="";
    String recorrDer="";
    if (this.hizq != null) {
      recorrIzq = this.getHizq().preorden();
    }
    if (this.hder != null) {
      recorrDer = this.getHder().preorden();
    }
    return (String.valueOf(this.valor)+" "+recorrIzq+" "+recorrDer).replaceAll("\\s+", " ");
  }

  public String inorden(){
    String recorrIzq="";
    String recorrDer="";
    if (this.hizq != null) {
      recorrIzq = this.getHizq().inorden();
    }
    if (this.hder != null) {
      recorrDer = this.getHder().inorden();
    }
    return (recorrIzq+" "+String.valueOf(this.valor)+" "+recorrDer).replaceAll("\\s+", " ");
  }  

  public String posorden(){
    String recorrIzq="";
    String recorrDer="";
    if (this.hizq != null) {
      recorrIzq = this.getHizq().posorden();
    }
    if (this.hder != null) {
      recorrDer = this.getHder().posorden();
    }
    return (recorrIzq+" "+recorrDer+" "+String.valueOf(this.valor)).replaceAll("\\s+", " ");
  }  

  public int getValor() {
    return valor;
  }

  public void setValor(int valor) {
    this.valor = valor;
  }

  public Arbol getHizq() {
    return hizq;
  }

  public void setHizq(Arbol hizq) {
    this.hizq = hizq;
  }

  public Arbol getHder() {
    return hder;
  }

  public void setHder(Arbol hder) {
    this.hder = hder;
  }

  
  

}

File: 2023-I/corte2/4-Arboles/ArbolesPython/Arbol.py
"""
27 de Mayo de 2023
Implementación arbol en Python
"""

class Arbol:
    def __init__(self, valor, hizq, hder):
      self.valor = valor
      self.hizq = hizq
      self.hder = hder

    def preorden(self):
      recorrIzq = ""
      recorrDer = ""
      if self.hizq is not None:
        recorrIzq = self.hizq.preorden()
      if self.hder is not None:
        recorrDer = self.hder.preorden()
      return (str(self.valor)+" "+recorrIzq+" "+recorrDer).strip()

    def inorden(self):
      recorrIzq = ""
      recorrDer = ""
      if self.hizq is not None:
        recorrIzq = self.hizq.inorden()
      if self.hder is not None:
        recorrDer = self.hder.inorden()
      return (recorrIzq+" "+str(self.valor)+" "+recorrDer).strip()
      
    def posorden(self):
      recorrIzq = ""
      recorrDer = ""
      if self.hizq is not None:
        recorrIzq = self.hizq.posorden()
      if self.hder is not None:
        recorrDer = self.hder.posorden()
      return (recorrIzq+" "+recorrDer +" "+str(self.valor)).strip()
